bigquery_etl/backfill/shredder_mitigation.py

```diff
@@ -0,0 +1,193 @@
+"""Generate a query to backfill an aggregate with shredder mitigation."""
+
+from datetime import date, datetime, time, timedelta
+from enum import Enum
+from types import NoneType
+
+TEMP_DATASET = "tmp"
+SUFFIX = datetime.now().strftime("%Y%m%d%H%M%S")
+PREVIOUS_DATE = (datetime.now() - timedelta(days=1)).strftime("%Y-%m-%d")
+
+
+class ColumnType(Enum):
+    """Differentiate metric and dimensions."""
+
+    METRIC = "METRIC"
+    DIMENSION = "DIMENSION"
+    UNDETERMINED = "None"
+
+
+class ColumnStatus(Enum):
+    """Different status of a column during shredder mitigation."""
+
+    COMMON = "COMMON"
+    ADDED = "ADDED"
+    REMOVED = "REMOVED"
+    UNDETERMINED = "None"
+
+
+class DataTypeGroup(Enum):
+    """Data types in BigQuery. Not including ARRAY and STRUCT as these are not expected in aggregate tables."""
```

Comment:
This grouping is mostly for debugging purposes. If a backfill with shredder mitigation fails, the output / logs will return information about the columns used, failed, data type group, if it's a new or existing column and if it's a metric or a dimension. That's why a general group works for starters.
It's also possible to remove the grouping and return the exact type in case a message like `Checks are failing due to a new column X of a **numeric** type used as a dimension in the query.` doesn't give enough details. Open to suggestions.